### ISSUE TITLE
Nested optimization for hierarchical search

### ISSUE THREAD

**Issue by oskar-taubert** [Helmholtz-AI-Energy/propulate#31]
The signature of the evaluation loss function and communication scheme should allow for arbitrary nesting of evolutionary processes. This allows for more efficient exploration of sub-spaces of different levels of variation cost.
E.g. long self supervised pre-training in an outer loop and fast finetuning in an inner loop.
